### PR TITLE
Added purple-bio 0.4.2.1 to recipes

### PR DESCRIPTION
**Pull request by TheBready**
:information_source:
If this is your first pull request to Bioconda please ping `@bioconda/core` so it can be reviewed and to request being added as a member of the Bioconda team. Members of the Bioconda team are able to merge their own pull requests once tests and linting have passed.

----------------

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by TheBready** [comment]
@bioconda/core 
This is my first recipe. Please review and let me know if I need to change something.

File: recipes/purple-bio/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv "
+  noarch: python
+
+requirements:
+  host:
+    - biopython
+    - pip
+    - python >=3
+    - tqdm
+    - pyyaml
```
Comment (dpryan79) [inline recipes/purple-bio/meta.yaml]: Can you try removing everything except `python` and `pip` from the host section? They're usually not needed for the actual installation.

File: recipes/purple-bio/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+test:
+  imports:
+    - purple
+
+about:
+  home: https://gitlab.com/HartkopfF/Purple
+  license: GNU Lesser General Public v3 or later (LGPLv3+)
+  license_family: LGPL
+  license_file:
```
Comment (pirovc) [inline recipes/purple-bio/meta.yaml]: maybe remove those?

File: recipes/purple-bio/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    - purple
+
+about:
+  home: https://gitlab.com/HartkopfF/Purple
+  license: GNU Lesser General Public v3 or later (LGPLv3+)
+  license_family: LGPL
+  license_file: 
+  summary: Picking Unique Relevant Peptides for viraL Experiments
+  doc_url: 
+  dev_url:
```
Comment (dpryan79) [inline recipes/purple-bio/meta.yaml]: Please remove the `doc_url:`, `dev_url:` and `license_file` entries.

**Comment by dpryan79** [comment]
Looks good! Only a few minor changes needed.

**Comment by TheBready** [comment]
Added all requested changes. Thanks.

**Comment by dpryan79** [comment]
Merged, thanks @TheBready!

**Pull request merged**